api/base.go

```diff
@@ -62,6 +63,33 @@ func (h *BaseHandler) Sorted(ctx *gin.Context) (db *gorm.DB) {
 	return
 }
 
+//
+// WithCount report count.
+// Sets the X-Total header for pagination.
+// Returns an error when count exceeds the limited and
+// is not constrained by pagination.
+func (h *BaseHandler) WithCount(ctx *gin.Context, count int64) (err error) {
+	n := int(count)
+	max := 500
+	p := Page{}
+	p.With(ctx)
+	if n > max {
+		if p.Limit == 0 || p.Limit > max {
+			err = &BadRequestError{
+				fmt.Sprintf(
+					"Found=%d, ?Limit <= %d required.",
+					n,
+					max)}
+			return
+		}
+	}
+	mp := ctx.Writer.Header()
+	mp[Total] = []string{
+		strconv.Itoa(int(count)),
+	}
+	return
+}
+
```

Comment:
This is needed ^ to prevent caller from making a request that will return 500K+ resources and blowing up the hub.

Reply:
500 seems like plenty to be requesting at once.